### ISSUE TITLE
[3.3.5] DB/Quest: shaman repeatable quest 1463, Earth Sapta

### ISSUE THREAD

**Issue by tkrokli** [TrinityCore/TrinityCore#17334]
**Description:**
- http://wow.gamepedia.com/Quest:Earth_Sapta_(Durotar)

While doing the Shaman quest line from level 1 to 5, I have discovered a couple of issues, the item page text issue is already reported in #17329. This time it is about Quest ID 1463, Earth Sapta - http://www.wowhead.com/quest=1463/earth-sapta / http://wotlk.openwow.com/quest=1463 and the quest giver Canaga Earthcaller (NPC entry 5887) http://wotlk.openwow.com/npc=5887 / http://www.wowhead.com/npc=5887/canaga-earthcaller offering this quest too soon, the very moment when quest 1517 is accepted and the player just received item 6635, Earth Sapta.

**Current behaviour:** Tell us what happens

At the very moment your shaman (level 4-5 character) accepts [Quest 1517, Call of Earth (2 of 3)](http://wotlk.openwow.com/quest=1517), [Canaga Earthcaller](http://wotlk.openwow.com/npc=5887) offers quest 1463 with missing text in the quest dialog.

![wowscrnshot_061116_023748](https://cloud.githubusercontent.com/assets/4903600/15982282/ee163d64-2f83-11e6-8bb5-9748962a387a.jpg)

**Expected behaviour:** Tell us what should happen instead

Canaga Earthcaller should not offer the quest 1463 unless item 6635 is missing from the player's bags or bank (http://www.wowhead.com/item=6635/earth-sapta )

**Steps to reproduce the problem:**

1. Create a level 4 or 5 orc or troll Shaman character.
2. Obtain quest 1517 (via gm command `.quest add 1517` or doing all level 1-4 quests)
3. Notice that Canaga Earthcaller immediately offers quest 1463 when quest 1517 is accepted.

**Branch(es)**: **3.3.5** (the quest was removed from World of Warcraft in patch 4.0.3a.)
http://wow.gamepedia.com/Quest:Earth_Sapta_(Durotar)

**TC hash/commit:**  TrinityCore rev. 2db9269 2016-06-10 20:03:30 +0100 (3.3.5 branch) (Win64, Release)

**TDB version:**  TDB 335.61 + updates up to and including `2016_06_09_06_world.sql`

**Operating system:**  Windows 10 Home x64 Version 1511 (OS Build 10586.318)

**Comment by tkrokli** [comment]
My bad, I did not take my time to check what was actually going on.

Canaga Earthcaller (NPC entry 5887) http://wotlk.openwow.com/npc=5887 just offers Quest 1463 (http://wotlk.openwow.com/quest=1463 / http://www.wowhead.com/quest=1463/earth-sapta) in case the player needs a new Earth Sapta (item 6635, http://www.wowhead.com/item=6635/earth-sapta) if the player has lost or thrown away the item instead of completing the quest.

I think we need a condition to make Canaga Earthcaller offer that repeatable quest **only if player does not have item 6635 in bags or bank** to not see the yellow **?** at the very moment quest 1517 is taken, as well as filling in the missing text so the "repeatable" quest can be explained properly. Check also how it looks on this page http://wotlk.openwow.com/quest=1463 and see that the quest text is missing.

**Comment by SnapperRy** [comment]
This applies for almost all the quests in #17255 too, but I wasn't able to find any source on whether the quests become unavailable if you already have the item you'd get a replacement of. It makes sense, but doubt that's enough :P

**Comment by tkrokli** [comment]
**Suggested solution** to make the quest visible only if player has lost item 6635, also adding the correct CompletionText to the quest so it contains some text for the player to read. CompletionText:
```
I give you one in good faith. You already proved yourself once, but me tinkin' you should be more careful in the future.
```
Source(s): http://www.wowhead.com/quest=1463/earth-sapta / http://wow.gamepedia.com/Quest:Earth_Sapta_(Durotar)
```sql
-- CompletionText for shaman class quest 1463, Earth Sapta (item replacement)
UPDATE `quest_request_items` SET `CompletionText`="I give you one in good faith. You already proved yourself once, but me tinkin' you should be more careful in the future." WHERE `ID`= 1463;
-- condition for shaman class quest 1463, Earth Sapta
DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId` IN (19,20) AND `SourceGroup`= 0 AND `SourceEntry`= 1463;
INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`SourceId`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionTarget`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`NegativeCondition`,`ErrorType`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
(19,0,1463,0,1,9,0, 1517,0,0 0,0,0, "","Accept quest 'Earth Sapta' if player has taken quest 1517, Call of Earth 2 of 3 (Durotar) - AND"),
(20,0,1463,0,1,2,0, 6635,1,1,1,0,0, "","Show quest mark for quest 'Earth Sapta' if player does NOT have item 6635 in bags or bank - AND");
```
I have not fully tested this theory yet, so the conditions might not work as intended. Feel free to test.

**Comment by tkrokli** [comment]
Oh. I did not anticipate that my issue contained enough information to actually create a solution based on my lacking information and testing. I did test some of this before 14:00 CEST, but unfortunately I had an appointment to spend the day on a local airshow, so now that I came back home just an hour ago, I did not think this issue would be closed yet with a solution which I had not decided on or tested enough.

I hope you are quite sure that the extra changes will make the commit work, but I will test it to see if it works now. Thanks for following up this, but if it looks like it still does not work, I can make a new issue with a new suggested solution. Sorry about the mess I made, I was tired and did not plan it well enough.
